tools/coredump/ebpfcode.go

```diff
@@ -12,8 +12,16 @@ package main
 // Also the tail call helper "bpf_tail_call" is overridden here, as it works in
 // tandem with the main entry point's setjmp.
 
+// todo: enable -Wunused-parameter after 1.25 upgrade https://go-review.googlesource.com/c/go/+/642196
+
 /*
-#define TESTING
+#cgo CFLAGS: -Wall -Wextra -Werror
+#cgo CFLAGS: -Wno-address-of-packed-member
+#cgo CFLAGS: -Wno-unused-label
+#cgo CFLAGS: -Wno-sign-compare
+#cgo CFLAGS: -Wno-unknown-pragmas
+#cgo CFLAGS: -Wno-unused-parameter
```

Comment:
Unfortunately  can not enable this one in `cgo` because of https://github.com/golang/go/issues/71225